src/promptflow-core/promptflow/_utils/async_utils.py

```diff
@@ -109,3 +111,29 @@ async def wrapper(*args, **kwargs):
             return await asyncio.get_event_loop().run_in_executor(executor, partial_func)
 
     return wrapper
+
+
+async def consume_from_queue_async(queue: Queue):
+    while True:
+        if queue.qsize() == 0:
+            #  If the queue is empty, wait for a while and try again.
+            await asyncio.sleep(0.1)
+            continue
```

Comment:
Do we need to set a max timeout here? If no one push items to the queue for a long time, will we wait forever?

Reply:
No timeout is added, it would keep the current behavior that it would wait forever if the generator is not finished.